server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -207,6 +210,56 @@ static TopDocs mergeTopDocs(Collection<TopDocs> results, int topN, int from) {
         return mergedTopDocs;
     }
 
+    private static void assertSameSortTypes(Collection<TopDocs> results, SortField[] firstSortFields) {
```

Comment:
Can we call this 'check' rather than 'assert'? Assert implies that it would be disabled if assertions are turned off.